src/fparser/two/Fortran2008.py

```diff
@@ -817,6 +819,82 @@ def alloc_opt_list(cls):
         return Alloc_Opt_List
 
 
+class Loop_Control(Loop_Control_2003):  # R818
+    """
+    Fortran 2008 rule R818
+
+    loop-control is [ , ] do-variable = scalar-int-expr , scalar-int-expr
+                       [ , scalar-int-expr ]
+                    or [ , ] WHILE ( scalar-logical-expr )
+                    or [ , ] CONCURRENT forall-header
+
+    Extends the Fortran2003 rule R830 with the additional CONCURRENT clause.
+
+    """
+
+    subclass_names = []
+    use_names = [
+        "Do_Variable",
```

Comment:
Unless I'm missing something, only `Forall_Header` is actually referenced in the implementation of this class?

Reply:
It is because we make use of the Fortran2003 match method so must include any classes used there as well. I've changed it so that we copy the Fortran2003 use_names values and append Forall_Header to them and have added a comment explaining. Hopefully that is clearer.

Reply:
Thanks. I don't *think* we need to do this but it doesn't hurt.